pilot/test/integration/cloudfoundry/copilot_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration_test
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"code.cloudfoundry.org/copilot/api"
+	"code.cloudfoundry.org/copilot/testhelpers"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"google.golang.org/grpc"
+	"istio.io/istio/pilot/model"
+	"istio.io/istio/pilot/platform/cloudfoundry"
+	"istio.io/istio/pilot/proxy/envoy"
+	"istio.io/istio/pilot/test/integration/cloudfoundry/mock"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	appPort           = 9000
+	pilotPort         = 5555
+	copilotAddr       = "127.0.0.1:5000"
+	pilotAddr         = "127.0.0.1:5555"
+	registrationRoute = "http://127.0.0.1:5555/v1/registration"
+	listenersRoute    = "http://127.0.0.1:5555/v1/listeners/x/router~x~x~x"
+	appEnvoyListener  = "http://127.0.0.1:6666"
+)
+
+func TestLaunchPilot(t *testing.T) {
```

Comment:
Can we give this a more descriptive name?   Maybe `TestEdgeRouterWithMockCopilot`?

Reply:
@rosenhouse I'd prefer to just land this rather than change the name / go through another test cycle.